pkg/pipeline/resource_pipeline_artifactory_integration.go

```diff
@@ -0,0 +1,138 @@
+package pipeline
+
+import (
+	"context"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/jfrog/terraform-provider-shared/util"
+)
+
+func pipelineArtifactoryProjectIntegrationResource() *schema.Resource {
+
+	var artifactoryIntegrationSchema = util.MergeSchema(
+		projectIntegrationSchema,
+		map[string]*schema.Schema{
+			"apikey": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Sensitive:    true,
+				ValidateFunc: validation.StringIsNotEmpty,
+				Description:  "ApiKey for Artifactory access",
+			},
+			"user": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+				Description:  "User for Artifactory access",
+			},
+			"url": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+				Description:  "URL to Artifactory instance",
+			},
+			"master_integration_id": {
+				Type:        schema.TypeInt,
+				Default:     98,
+				Optional:    true,
+				Description: "The Id of the master integration.",
+			},
+			"master_integration_name": {
+				Type:        schema.TypeString,
+				Default:     "artifactory",
+				Optional:    true,
+				Description: "The name of the master integration.",
+			},
+		},
+	)
+
+	var unpackArtifactoryFormValues = func(data *schema.ResourceData) []FormJSONValues {
+		d := &util.ResourceData{data}
+		var formJSONValues = []FormJSONValues{
+			{
+				Label: "apikey",
+				Value: d.GetString("apikey", true),
+			},
+			{
+				Label: "url",
+				Value: d.GetString("url", false),
+			},
+			{
+				Label: "user",
+				Value: d.GetString("user", false),
+			},
+		}
+		return formJSONValues
+	}
+
+	var packArtifactoryFormValues = func(d *schema.ResourceData, formJSONValues []FormJSONValues) []error {
+		setValue := util.MkLens(d)
+		var errors []error
+		for _, idx := range formJSONValues {
```

Comment:
The use of `idx` as name is confusing here. `_` is the index from `for` loop so we should use `jsonValue` or something similar instead.

Reply:
corrected